src/mono/mono/mini/mini-amd64.c

```diff
@@ -1990,13 +1990,18 @@ mono_arch_allocate_vars (MonoCompile *cfg)
 			}
 			case ArgSwiftError: {
 					ins->opcode = OP_REGOFFSET;
-					ins->inst_basereg = cfg->frame_reg;
-					ins->inst_offset = ainfo->offset + ARGS_OFFSET;
 					offset = ALIGN_TO (offset, sizeof (target_mgreg_t));
+					ins->inst_basereg = cfg->frame_reg;
+					ins->inst_offset = offset;
```

Comment:
Why it doesn't include the `ARGS_OFFSET` offset between fp and the first argument in the callee?

Reply:
We don't enforce any signature rules, either. The Swift special register types can appear anywhere in the signature's parameter list.

Reply:
@kotlarmilos i think for that to work with ainfo->offset without any modifications, the error would have to be the first argument. We could probably update ainfo->offset in get_call_info so that it works for every position. I'm not sure what are the benefits of using ainfo->offset though.

Reply:
Ok, sounds good. I forgot that the `ainfo->offset` is always 0 in the `get_call_info`.